iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.inl

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_POSH_INTERNAL_ROUDI_PORT_MANAGER_INL
+#define IOX_POSH_INTERNAL_ROUDI_PORT_MANAGER_INL
+
+namespace iox
+{
+namespace roudi
+{
+template <typename T, cxx::enable_if_t<std::is_same<T, iox::build::OneToManyPolicy>::value>*>
+inline bool PortManager::hasDuplicatePublisher(const capro::ServiceDescription& service,
+                                               const ProcessName_t& processName) noexcept
+{
+    // check if the publisher is already in the list
+    for (auto publisherPortData : m_portPool->getPublisherPortDataList())
+    {
+        popo::PublisherPortRouDi publisherPort(publisherPortData);
+        if (service == publisherPort.getCaProServiceDescription())
+        {
+            LogWarn() << "Process '" << processName
```

Comment:
I would move the error handling outside of this method into the method which is calling `hasDuplicatePublishers`. The reason is that no one would expect that a call to this method breaks the whole system. Maybe you would like to know in an 1 to m environment if this publisher is already created and if it is not created you would create a publisher.

But at the moment you would ask this method in an innocent way and would crash for no reasons from your point of view when there is already a publisher which is not an error! It becomes only then an error if you create one.

I would call the error handler exactly where the error has the effect of screwing up the system. This is not the place. The place is the method where this method is called.

Reply:
Got your point. My approach was to included as few code in the text segment as possible for both variants. Other opinions? @elBoberido @budrus

Reply:
I'm with @elfenpiff. 
If you rename it like I proposed, yopu would end up in something like
`if(violatesCommunicationPoliy())
{errorHandler(Bäääm);
}`

Reply:
Done